src/relayclient/RelayClient.ts

```diff
@@ -162,15 +164,25 @@ export class RelayClient {
     return false
   }
 
-  async _init (): Promise<void> {
-    if (this.initialized) { return }
-    this.emit(new GsnInitEvent())
-    await this.contractInteractor.init()
-    this.initialized = true
+  async init (): Promise<this> {
+    await this.initMutex.runExclusive(async () => {
+      if (this.initialized) {
+        return this
+      }
+      this.emit(new GsnInitEvent())
+      await this.contractInteractor.init()
+      this.initialized = true
+    })
+    return this
   }
 
   async relayTransaction (gsnTransactionDetails: GsnTransactionDetails): Promise<RelayingResult> {
-    await this._init()
+    if (!this.initialized) {
+      if (!this.initMutex.isLocked()) {
+        this._warn('better call RelayClient.init() in advance (to make first request faster)')
```

Comment:
How big of a deal it would be to just enforce it? I see very little value in keeping the old flow possible.